kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/ingest/dumper/InventoryDumper.java

```diff
@@ -188,9 +190,12 @@ private DataRecord loadDataRecord(final ResultSet resultSet, final ResultSetMeta
         DataRecord result = new DataRecord(newPosition(resultSet), columnCount);
         result.setType(IngestDataChangeType.INSERT);
         result.setTableName(dumperConfig.getLogicTableName());
+        List<String> insertColumnNames = Optional.ofNullable(dumperConfig.getInsertColumnNames()).orElse(Collections.emptyList());
```

Comment:
Could we verify `resultSetMetaData.getColumnCount()` is equals with `dumperConfig.getInsertColumnNames().size`?

Reply:
<img width="785" alt="image" src="https://user-images.githubusercontent.com/101622833/230863072-3a5f24f8-13b8-43d4-bc79-00a1da60afed.png">
The `resultSetMetaData.getColumnCount()` depends on querySQL, if provider querySQL, must specify `dumperConfig.getInsertColumnNames()`,  is there a better way to handle it?

Reply:
Just verify it when `dumperConfig.getInsertColumnNames()` is available